src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,74 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+/**
+ * Loads the task list from a file in the hard disk and updates any changes in the
+ * task list in the same file
+ */
+public class Storage {
+
+    private final String filename;
+
+    public Storage(String filename) {
+        this.filename = filename;
+    }
+
+    /**
+     * Loads list of tasks previously saved in hard disk
+     * If no previous file exists, create a new file
+     * @return ArrayList of tasks that were loaded from hard disk
+     * @throws IOException
+     */
+    public ArrayList<Task> loadTasks() throws IOException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        File file = new File(this.filename);
+        file.createNewFile();
+        Scanner fileScanner = new Scanner(file);
+        while (fileScanner.hasNext()) {
+            String tempTask = fileScanner.nextLine();
+            String[] loadedTask = tempTask.split("~~");
+            switch (loadedTask[0]) {
+            case "T":
+                tasks.add(new Todo(loadedTask[2]));
+                break;
+            case "D":
+                tasks.add(new Deadline(loadedTask[2], loadedTask[3]));
+                break;
+            case "E":
+                tasks.add(new Event(loadedTask[2], loadedTask[3]));
+                break;
+            default:
+                break;
+            }
+            if (loadedTask[1].equals("X")) {
```

Comment:
Perhaps you could name loadTask[0] and loadTask[1], e.g. taskType and isDone or hasMarked, to improve readability. 🤔 😃